src/routes/data.rs

```diff
@@ -217,3 +218,16 @@ pub async fn retrieve_card(
 
     Ok(Json(card.try_into()?))
 }
+
+/// `/cards/fingerprint` handling the creation and retrieval of card fingerprint
+pub async fn get_or_insert_fingerprint(
+    extract::State(state): extract::State<AppState>,
+    Json(request): Json<types::FingerprintRequest>,
+) -> Result<Json<types::FingerprintResponse>, ContainerError<error::ApiError>> {
+    let fingerprint = state
+        .db
+        .insert_fingerprint(request.card_number, request.hash_key)
+        .await?;
```

Comment:
Also, can we have a validation check for the card number, similar to the one that we have when storing cards?

[src/routes/data/types.rs:123:35](https://github.com/juspay/hyperswitch-card-vault/blob/bf57a3c182786fbf692eb38776730f9a9186af41/src/routes/data/types.rs#L123-L134)